riscv/encoding.h

```diff
@@ -3132,6 +3132,35 @@ DECLARE_CSR(mhpmcounter29h, CSR_MHPMCOUNTER29H)
 DECLARE_CSR(mhpmcounter30h, CSR_MHPMCOUNTER30H)
 DECLARE_CSR(mhpmcounter31h, CSR_MHPMCOUNTER31H)
 #endif
+#ifdef CSR_DECLARE
```

Comment:
`encoding.h` is generating by `riscv-opcode`, don't edit it directly, and seems like `CSR_DECLARE` is similar to `DECLARE_CSR`, but a small subset, I guess the reason is want to prevent `sim_t::interactive_csr` dump all CSR registers?

Reply:
Exactly, call get_csr() with some uninplemented csr may cause a illegal instrucions trap so i want to skip them.